processor/src/main/java/io/jonasg/bob/BuilderTypeSpecFactory.java

```diff
@@ -116,37 +116,58 @@ private List<MethodSpec> generateSetters() {
 	}
 
 	protected MethodSpec generateSetterForField(BuildableField field) {
+
 		var builder = MethodSpec.methodBuilder(setterName(field.fieldName()))
 				.addModifiers(Modifier.PUBLIC)
 				.returns(builderType())
 				.addParameter(TypeName.get(field.type()), field.fieldName());
-		if (field.isConstructorArgument() && isEnforcedConstructorPolicy() || field.isMandatory()) {
+		if (field.isConstructorArgument() && isAnEnforcedConstructorPolicy() || field.isMandatory()) {
 			builder.addStatement("this.$L.set($L)", field.fieldName(), field.fieldName());
+			if (isEnforcedAllowNullsConstructorPolicy()) {
+				builder.addStatement("this.fieldSet = true");
+			}
 		} else {
 			builder.addStatement("this.$L = $L", field.fieldName(), field.fieldName());
 		}
 		return builder.addStatement("return this")
 				.build();
 	}
 
+	private boolean isAnEnforcedConstructorPolicy() {
```

Comment:
rename to `isEnforcedConstructorPolicy` no need for `An`

Reply:
I was thinking for in case there is need to only check for constructorPolicy `ENFORCED,` then the name is already taken by a check on both